client/gtk2/ibusimcontext.c

```diff
@@ -1529,6 +1530,69 @@ _set_cursor_location_internal (IBusIMContext *ibusimcontext)
 
     area = ibusimcontext->cursor_area;
 
+#if GTK_CHECK_VERSION (3, 98, 4)
+    graphene_point_t p;
+
+    if (!gtk_widget_compute_point (ibusimcontext->client_window,
+                                   GTK_WIDGET (gtk_widget_get_root (ibusimcontext->client_window)),
+                                   &GRAPHENE_POINT_INIT (area.x, area.y),
+                                   &p))
+        graphene_point_init (&p, area.x, area.y);
+
+    gtk_native_get_surface_transform (gtk_widget_get_native (ibusimcontext->client_window), &nx, &ny);
+
+    display = gtk_widget_get_display (ibusimcontext->client_window);
+    if (GDK_IS_X11_DISPLAY (display)) {
+        GdkSurface *surface = gtk_native_get_surface
+            (gtk_widget_get_native (ibusimcontext->client_window));
+        GdkX11Surface *impl = GDK_X11_SURFACE (surface);
+        Window child;
+        int scale_factor = gtk_widget_get_scale_factor
+            (ibusimcontext->client_window);
+
+        XTranslateCoordinates (GDK_SURFACE_XDISPLAY (surface),
+                               GDK_SURFACE_XID (surface),
+                               gdk_x11_display_get_xrootwindow (display),
+                               0, 0, &tx, &ty,
+                               &child);
+
+        tx = tx / scale_factor;
+        ty = ty / scale_factor;
+    }
+
+    area.x = p.x + nx + tx;
+    area.y = p.y + ny + ty;
+#else
+    gdk_window_get_root_coords (ibusimcontext->client_window,
+                                area.x, area.y,
+                                &area.x, &area.y);
+#endif
+
+    _set_rect_scale_factor_with_window (&area, ibusimcontext->client_window);
+
+#if !GTK_CHECK_VERSION (3, 98, 4)
+    display = gdk_window_get_display (ibusimcontext->client_window);
+#endif
+
+#ifdef GDK_WINDOWING_WAYLAND
+    if (GDK_IS_WAYLAND_DISPLAY (display))
+        ibus_input_context_set_cursor_location_relative (ibusimcontext->ibuscontext,
```

Comment:
Since the `ibus_input_context_set_cursor_location_relative()` is described with multiple lines, please enclose it with `{}`.

```
+    if (GDK_IS_WAYLAND_DISPLAY (display)) {
+        ibus_input_context_set_cursor_location_relative (
+                ibusimcontext->ibuscontext,
+                area.x,
+                area.y,
+                area.width,
+                area.height);
 
-    else
+    } else {
```